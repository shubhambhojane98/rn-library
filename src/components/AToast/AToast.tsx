import React, {useEffect, useRef} from 'react';
import {Animated, StyleSheet, View} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import ATypography from '../../components/ATypography/ATypography';
import {TypographyVariant} from '../../components/ATypography/ATypographyEnum';
import IconSVG from '../../assets/svgs';
import {defaultScale} from '../../utils/Common';
import {Color} from '../../theme';

interface Props {
  label: string;
  toastTime?: number;
  iconName?: any;
  iconHeight?: number;
  iconWidth?: number;
  color?: string;
  backgroundColor?: string;
}
const defaultSize = moderateScale(20, defaultScale);

const AToast: React.FC<Props> = ({
  label,
  toastTime,
  iconName,
  iconHeight,
  iconWidth,
  color,
  backgroundColor,
}) => {
  const opacity = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    Animated.sequence([
      Animated.timing(opacity, {
        toValue: 1,
        duration: 500,
        useNativeDriver: true,
      }),
      Animated.delay(toastTime ? toastTime : 2000),
      Animated.timing(opacity, {
        toValue: 0,
        duration: 500,
        useNativeDriver: true,
      }),
    ]).start();
  }, []);

  return (
    <View style={styles.container}>
      <Animated.View
        style={{
          ...styles.animatedView,
          opacity,
          backgroundColor,
          transform: [
            {
              translateY: opacity.interpolate({
                inputRange: [0, 1],
                outputRange: [-20, 0],
              }),
            },
          ],
        }}>
        {iconName ? (
          <IconSVG
            name={iconName}
            height={moderateScale(iconHeight || defaultSize, defaultScale)}
            width={moderateScale(iconWidth || defaultSize, defaultScale)}
          />
        ) : null}
        <ATypography
          children={label}
          variant={TypographyVariant.SECONDARY}
          color={color}
          style={styles.text}
        />
      </Animated.View>
    </View>
  );
};

export default AToast;

const styles = StyleSheet.create({
  container: {
    bottom: moderateScale(20, defaultScale),
    position: 'absolute',
    zIndex: 1,
    left: moderateScale(10, defaultScale),
    right: moderateScale(10, defaultScale),
  },
  animatedView: {
    flexDirection: 'row',
    padding: moderateScale(10, defaultScale),
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'black',
    borderRadius: moderateScale(4, defaultScale),
    height: moderateScale(64, defaultScale),
    width: moderateScale(320, defaultScale),
    alignSelf: 'center',
  },
  text: {
    paddingHorizontal: moderateScale(10, defaultScale),
  },
});

AToast.defaultProps = {
  color: Color.white,
  backgroundColor: Color.black,
};
