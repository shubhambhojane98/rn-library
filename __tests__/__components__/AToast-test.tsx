import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AToast from '../../src/components/AToast/AToast';

beforeEach(() => {
  jest.useFakeTimers();
});

it('renders correctly with label', () => {
  const tree = renderer
    .create(<AToast label={'Successfully verified your phone number'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props', () => {
  const tree = renderer
    .create(
      <AToast
        iconName={'checkboxfilled'}
        label={'Successfully verified your phone number'}
        toastTime={1000}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
