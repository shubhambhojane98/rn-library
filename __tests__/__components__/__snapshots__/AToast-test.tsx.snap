// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<View
  style={
    Object {
      "bottom": 24.57142857142857,
      "left": 12.285714285714285,
      "position": "absolute",
      "right": 12.285714285714285,
      "zIndex": 1,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#231F20",
        "borderRadius": 4.914285714285715,
        "flexDirection": "row",
        "height": 78.62857142857143,
        "justifyContent": "center",
        "opacity": 0,
        "padding": 12.285714285714285,
        "transform": Array [
          Object {
            "translateY": -20,
          },
        ],
        "width": 393.1428571428571,
      }
    }
  >
    <SvgMock
      height={30.187755102040814}
      name="checkboxfilled"
      width={30.187755102040814}
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "AvenirNextLTPro-Regular",
            "fontSize": 21.13142857142857,
            "fontWeight": "normal",
          },
          Object {
            "color": "#fff",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "paddingHorizontal": 12.285714285714285,
          },
        ]
      }
    >
      Successfully verified your phone number
    </Text>
  </View>
</View>
`;

exports[`renders correctly with label 1`] = `
<View
  style={
    Object {
      "bottom": 24.57142857142857,
      "left": 12.285714285714285,
      "position": "absolute",
      "right": 12.285714285714285,
      "zIndex": 1,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#231F20",
        "borderRadius": 4.914285714285715,
        "flexDirection": "row",
        "height": 78.62857142857143,
        "justifyContent": "center",
        "opacity": 0,
        "padding": 12.285714285714285,
        "transform": Array [
          Object {
            "translateY": -20,
          },
        ],
        "width": 393.1428571428571,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "AvenirNextLTPro-Regular",
            "fontSize": 21.13142857142857,
            "fontWeight": "normal",
          },
          Object {
            "color": "#fff",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "paddingHorizontal": 12.285714285714285,
          },
        ]
      }
    >
      Successfully verified your phone number
    </Text>
  </View>
</View>
`;
